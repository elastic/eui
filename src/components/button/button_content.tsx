/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React, { HTMLAttributes, FunctionComponent, Ref } from 'react';
import classNames from 'classnames';
import { CommonProps, keysOf } from '../common';
import { EuiLoadingSpinner } from '../loading';
import { EuiIcon, IconType } from '../icon';

export type ButtonContentIconSide = 'left' | 'right';

const iconSideToClassNameMap: {
  [side in ButtonContentIconSide]: string | null;
} = {
  left: null,
  right: 'euiButtonContent--iconRight',
};

export const ICON_SIDES = keysOf(iconSideToClassNameMap);

export type EuiButtonContentType = HTMLAttributes<HTMLSpanElement>;

/**
 * *INTERNAL ONLY*
 * This component is simply a helper component for reuse within other button components
 */
export interface EuiButtonContentProps extends CommonProps {
  /**
   * Any `type` accepted by EuiIcon
   */
  iconType?: IconType;
  /**
   * Can only be one side `left` or `right`
   */
  iconSide?: ButtonContentIconSide;
  isLoading?: boolean;
  /**
   * Object of props passed to the <span/> wrapping the content's text/children only (not icon)
   */
  textProps?: HTMLAttributes<HTMLSpanElement> &
    CommonProps & {
      ref?: Ref<HTMLSpanElement>;
      'data-text'?: string;
    };
  iconSize?: 's' | 'm';
}

export const EuiButtonContent: FunctionComponent<
  EuiButtonContentType & EuiButtonContentProps
> = ({
  children,
  textProps,
  isLoading = false,
  iconType,
  iconSize = 'm',
  iconSide = 'left',
  ...contentProps
}) => {
  // Add an icon to the button if one exists.
  let buttonIcon;

  if (isLoading) {
    buttonIcon = (
      <EuiLoadingSpinner className="euiButtonContent__spinner" size="m" />
    );
  } else if (iconType) {
    buttonIcon = (
      <EuiIcon
        className="euiButtonContent__icon"
        type={iconType}
        size={iconSize}
        color="inherit" // forces the icon to inherit its parent color
      />
    );
  }

  const contentClassNames = classNames(
    'euiButtonContent',
    iconSide ? iconSideToClassNameMap[iconSide] : null,
    contentProps && contentProps.className
  );

  return (
    <span {...contentProps} className={contentClassNames}>
      {buttonIcon}
      <span {...textProps}>{children}</span>
    </span>
  );
};
