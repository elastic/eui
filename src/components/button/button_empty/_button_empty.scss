/**
 * 1. We don't want any of the animations that come inherited from the mixin.
 *    These should act like normal links instead.
 * 2. Change the easing, quickness to not bounce so lighter backgrounds don't flash
 */
.euiButtonEmpty {
  @include euiButton;

  border-color: transparent;
  background-color: transparent;
  box-shadow: none;
  // sass-lint:disable-block no-important
  transform: none !important; /* 1 */
  animation: none !important; /* 1 */
  transition-timing-function: ease-in; /* 2 */
  transition-duration: $euiAnimSpeedFast; /* 2 */

  .euiButtonEmpty__content {
    padding: 0 $euiSizeS;
  }

  .euiButtonEmpty__text {
    text-overflow: ellipsis;
    overflow: hidden;
  }

  &.euiButtonEmpty--small {
    height: $euiButtonHeightSmall;
  }

  &.euiButtonEmpty--xSmall {
    height: $euiButtonHeightXSmall;
    font-size: $euiFontSizeS;
  }

  &:disabled {
    @include euiButtonContentDisabled;

    color: $euiButtonColorDisabledText;

    &:focus {
      background-color: transparent;
    }

    &:hover,
    &:focus {
      text-decoration: none;
    }
  }
}


.euiButtonEmpty--flushLeft,
.euiButtonEmpty--flushRight,
.euiButtonEmpty--flushBoth {
  .euiButtonEmpty__content {
    padding-left: 0;
    padding-right: 0;
  }
}

.euiButtonEmpty--flushLeft {
  margin-right: $euiSizeS;
}

.euiButtonEmpty--flushRight {
  margin-left: $euiSizeS;
}

// Modifier naming and colors.
$euiButtonEmptyTypes: (
  primary: $euiColorPrimaryText,
  danger: $euiColorDangerText,
  disabled: $euiButtonColorDisabledText,
  ghost: $euiColorGhost,
  text: $euiTextColor,
  success: $euiColorSecondaryText,
  warning: $euiColorWarningText,
);

// Create button modifiers based upon the map.
@each $name, $color in $euiButtonEmptyTypes {
  .euiButtonEmpty--#{$name} {
    @if ($name == 'ghost') {
      // Ghost is unique and ALWAYS sits against a dark background.
      color: $color;
    } @else if ($name == 'text') {
      // The default color is lighter than the normal text color, make the it the text color
      color: $euiTextColor;
    } @else {
      // Other colors need to check their contrast against the page background color.
      color: makeHighContrastColor($color, $euiColorEmptyShade);
    }

    &:focus {
      background-color: transparentize($color, .9);
    }

    &:hover {
      @if ($name == 'disabled') {
        cursor: not-allowed;
      }
    }
  }
}

