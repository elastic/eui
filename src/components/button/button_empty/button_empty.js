import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import {
  checkHrefAndOnClick,
  getSecureRelForTarget,
} from '../../../services';

import {
  ICON_TYPES,
  EuiIcon,
} from '../../icon';

const colorToClassNameMap = {
  primary: 'euiButtonEmpty--primary',
  danger: 'euiButtonEmpty--danger',
  disabled: 'euiButtonEmpty--disabled',
  text: 'euiButtonEmpty--text',
  ghost: 'euiButtonEmpty--ghost',
};

export const COLORS = Object.keys(colorToClassNameMap);

const sizeToClassNameMap = {
  xs: 'euiButtonEmpty--xSmall',
  s: 'euiButtonEmpty--small',
  l: 'euiButtonEmpty--large',
};

export const SIZES = Object.keys(sizeToClassNameMap);

const iconSideToClassNameMap = {
  left: '',
  right: 'euiButtonEmpty--iconRight',
};

export const ICON_SIDES = Object.keys(iconSideToClassNameMap);

const flushTypeToClassNameMap = {
  'left': 'euiButtonEmpty--flushLeft',
  'right': 'euiButtonEmpty--flushRight',
};

export const FLUSH_TYPES = Object.keys(flushTypeToClassNameMap);

export const EuiButtonEmpty = ({
  children,
  className,
  iconType,
  iconSide,
  color,
  size,
  flush,
  isDisabled,
  href,
  target,
  rel,
  onClick,
  type,
  ...rest
}) => {

  const classes = classNames(
    'euiButtonEmpty',
    colorToClassNameMap[color],
    sizeToClassNameMap[size],
    iconSideToClassNameMap[iconSide],
    flushTypeToClassNameMap[flush],
    className,
  );

  // Add an icon to the button if one exists.
  let buttonIcon;

  if (iconType) {
    buttonIcon = (
      <EuiIcon
        className="euiButtonEmpty__icon"
        type={iconType}
        size="m"
        aria-hidden="true"
      />
    );
  }

  if (href) {
    const secureRel = getSecureRelForTarget(target, rel);

    return (
      <a
        className={classes}
        href={href}
        target={target}
        rel={secureRel}
        {...rest}
      >
        <span className="euiButtonEmpty__content">
          {buttonIcon}
          <span>{children}</span>
        </span>
      </a>
    );
  } else {
    return (
      <button
        disabled={isDisabled}
        className={classes}
        onClick={onClick}
        type={type}
        {...rest}
      >
        <span className="euiButtonEmpty__content">
          {buttonIcon}
          <span>{children}</span>
        </span>
      </button>
    );
  }
};

EuiButtonEmpty.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  iconType: PropTypes.oneOf(ICON_TYPES),
  iconSide: PropTypes.oneOf(ICON_SIDES),
  color: PropTypes.oneOf(COLORS),
  size: PropTypes.oneOf(SIZES),
  flush: PropTypes.oneOf(FLUSH_TYPES),
  isDisabled: PropTypes.bool,
  href: checkHrefAndOnClick,
  target: PropTypes.string,
  rel: PropTypes.string,
  onClick: PropTypes.func,
  type: PropTypes.string,
};

EuiButtonEmpty.defaultProps = {
  type: 'button',
  iconSide: 'left',
  color: 'primary',
};
