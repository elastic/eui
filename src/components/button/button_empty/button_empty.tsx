import React, { FunctionComponent, HTMLAttributes } from 'react';
import classNames from 'classnames';

import {
  CommonProps,
  ExclusiveUnion,
  PropsForAnchor,
  PropsForButton,
  keysOf,
} from '../../common';
import { EuiLoadingSpinner } from '../../loading';
import { getSecureRelForTarget } from '../../../services';
import { IconType, EuiIcon } from '../../icon';

const colorToClassNameMap = {
  primary: 'euiButtonEmpty--primary',
  danger: 'euiButtonEmpty--danger',
  disabled: 'euiButtonEmpty--disabled',
  text: 'euiButtonEmpty--text',
  ghost: 'euiButtonEmpty--ghost',
};

export const COLORS = keysOf(colorToClassNameMap);

const sizeToClassNameMap = {
  xs: 'euiButtonEmpty--xSmall',
  s: 'euiButtonEmpty--small',
  l: 'euiButtonEmpty--large',
};

export const SIZES = keysOf(sizeToClassNameMap);

const iconSideToClassNameMap = {
  left: '',
  right: 'euiButtonEmpty--iconRight',
};

export const ICON_SIDES = keysOf(iconSideToClassNameMap);

const flushTypeToClassNameMap = {
  left: 'euiButtonEmpty--flushLeft',
  right: 'euiButtonEmpty--flushRight',
};

export const FLUSH_TYPES = keysOf(flushTypeToClassNameMap);

export interface EuiButtonEmptyProps extends CommonProps {
  iconType?: IconType;
  iconSide?: keyof typeof iconSideToClassNameMap;
  color?: keyof typeof colorToClassNameMap;
  size?: keyof typeof sizeToClassNameMap;
  flush?: keyof typeof flushTypeToClassNameMap;
  isDisabled?: boolean;
  href?: string;
  target?: string;
  rel?: string;

  /**
   * Adds/swaps for loading spinner & disables
   */
  isLoading?: boolean;

  type?: 'button' | 'submit';
  buttonRef?: () => void;
  /**
   * Passes props to `euiButtonEmpty__content` span
   */
  contentProps?: Partial<HTMLAttributes<HTMLSpanElement>>;

  /**
   * Passes props to `euiButtonEmpty__text` span
   */
  textProps?: Partial<HTMLAttributes<HTMLSpanElement>>;
}

type EuiButtonEmptyPropsForAnchor = PropsForAnchor<EuiButtonEmptyProps>;

type EuiButtonEmptyPropsForButton = PropsForButton<EuiButtonEmptyProps>;

type Props = ExclusiveUnion<
  EuiButtonEmptyPropsForAnchor,
  EuiButtonEmptyPropsForButton
>;

export const EuiButtonEmpty: FunctionComponent<Props> = ({
  children,
  className,
  iconType,
  iconSide = 'left',
  color = 'primary',
  size,
  flush,
  isDisabled,
  isLoading,
  href,
  target,
  rel,
  type = 'button',
  buttonRef,
  contentProps,
  textProps,
  ...rest
}) => {
  // If in the loading state, force disabled to true
  isDisabled = isLoading ? true : isDisabled;

  const classes = classNames(
    'euiButtonEmpty',
    colorToClassNameMap[color],
    size ? sizeToClassNameMap[size] : null,
    iconSideToClassNameMap[iconSide],
    flush ? flushTypeToClassNameMap[flush] : null,
    className
  );

  const contentClassNames = classNames(
    'euiButtonEmpty__content',
    contentProps && contentProps.className
  );

  const textClassNames = classNames(
    'euiButtonEmpty__text',
    textProps && textProps.className
  );

  // Add an icon to the button if one exists.
  let buttonIcon;

  if (isLoading) {
    buttonIcon = <EuiLoadingSpinner className="euiButton__spinner" size="m" />;
  } else if (iconType) {
    buttonIcon = (
      <EuiIcon
        className="euiButtonEmpty__icon"
        type={iconType}
        size="m"
        aria-hidden="true"
      />
    );
  }

  const innerNode = (
    <span {...contentProps} className={contentClassNames}>
      {buttonIcon}
      <span {...textProps} className={textClassNames}>
        {children}
      </span>
    </span>
  );

  // <a> elements don't respect the `disabled` attribute. So if we're disabled, we'll just pretend
  // this is a button and piggyback off its disabled styles.
  if (href && !isDisabled) {
    const secureRel = getSecureRelForTarget({ href, target, rel });

    return (
      <a
        className={classes}
        href={href}
        target={target}
        rel={secureRel}
        ref={buttonRef}
        {...rest as EuiButtonEmptyPropsForAnchor}>
        {innerNode}
      </a>
    );
  }

  return (
    <button
      disabled={isDisabled}
      className={classes}
      type={type}
      ref={buttonRef}
      {...rest as EuiButtonEmptyPropsForButton}>
      {innerNode}
    </button>
  );
};
