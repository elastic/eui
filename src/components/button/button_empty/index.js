export {
  EuiButtonEmpty,
} from './button_empty';
