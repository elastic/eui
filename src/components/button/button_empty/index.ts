export {
  COLORS,
  ICON_SIDES,
  EuiButtonEmpty,
  EuiButtonEmptyColor,
  EuiButtonEmptyProps,
  EuiButtonEmptySizes,
} from './button_empty';
