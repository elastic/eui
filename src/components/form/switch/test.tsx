import React from 'react';

import { EuiSwitch } from './switch';

export const Test = ({ checked }: { checked: boolean }) => (
  <EuiSwitch
    checked={checked}
    label="Test"
    onChange={e => console.log(e.target.checked)}
  />
);
