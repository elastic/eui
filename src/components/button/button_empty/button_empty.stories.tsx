/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import type { Meta, StoryObj } from '@storybook/react';
import { disableStorybookControls } from '../../../../.storybook/utils';

import { EuiButtonEmpty, EuiButtonEmptyProps } from './button_empty';

const meta: Meta<EuiButtonEmptyProps> = {
  title: 'EuiButtonEmpty',
  component: EuiButtonEmpty,
  argTypes: {
    flush: {
      options: [undefined, 'left', 'right', 'both'],
    },
    iconType: { control: 'text' },
    target: { control: 'text' },
  },
  args: {
    // Component defaults
    type: 'button',
    color: 'primary',
    size: 'm',
    iconSize: 'm',
    iconSide: 'left',
    isDisabled: false,
    isLoading: false,
    isSelected: false,
  },
};

export default meta;
type Story = StoryObj<EuiButtonEmptyProps>;

export const Playground: Story = {
  argTypes: disableStorybookControls(['buttonRef']),
  args: {
    children: 'Tertiary action',
  },
};
