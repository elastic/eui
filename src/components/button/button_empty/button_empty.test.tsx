/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { render, mount } from 'enzyme';
import { requiredProps } from '../../../test/required_props';

import { EuiButtonEmpty, COLORS, SIZES, FLUSH_TYPES } from './button_empty';
import { ICON_SIDES } from '../button_content';

describe('EuiButtonEmpty', () => {
  test('is rendered', () => {
    const component = render(
      <EuiButtonEmpty {...requiredProps}>Content</EuiButtonEmpty>
    );

    expect(component).toMatchSnapshot();
  });

  describe('props', () => {
    describe('isDisabled', () => {
      it('is rendered', () => {
        const component = render(<EuiButtonEmpty isDisabled />);

        expect(component).toMatchSnapshot();
      });

      it('renders a button even when href is defined', () => {
        const component = render(<EuiButtonEmpty href="#" isDisabled />);

        expect(component).toMatchSnapshot();
      });

      it('renders if passed simply as disabled', () => {
        const component = render(<EuiButtonEmpty disabled />);

        expect(component).toMatchSnapshot();
      });
    });

    describe('isLoading', () => {
      it('is rendered', () => {
        const component = render(<EuiButtonEmpty isLoading />);

        expect(component).toMatchSnapshot();
      });
    });

    describe('isSelected', () => {
      it('is rendered as true', () => {
        const component = render(<EuiButtonEmpty isSelected />);

        expect(component).toMatchSnapshot();
      });

      it('is rendered as false', () => {
        const component = render(<EuiButtonEmpty isSelected={false} />);

        expect(component).toMatchSnapshot();
      });
    });

    describe('iconType', () => {
      it('is rendered', () => {
        const component = render(<EuiButtonEmpty iconType="user" />);

        expect(component).toMatchSnapshot();
      });
    });

    describe('color', () => {
      COLORS.forEach((color) => {
        test(`${color} is rendered`, () => {
          const component = render(<EuiButtonEmpty color={color} />);

          expect(component).toMatchSnapshot();
        });
      });
    });

    describe('size', () => {
      SIZES.forEach((size) => {
        test(`${size} is rendered`, () => {
          const component = render(<EuiButtonEmpty size={size} />);

          expect(component).toMatchSnapshot();
        });
      });
    });

    describe('iconSide', () => {
      ICON_SIDES.forEach((iconSide) => {
        test(`${iconSide} is rendered`, () => {
          const component = render(
            <EuiButtonEmpty iconType="user" iconSide={iconSide}>
              Content
            </EuiButtonEmpty>
          );

          expect(component).toMatchSnapshot();
        });
      });
    });

    describe('flush', () => {
      FLUSH_TYPES.forEach((flushType) => {
        test(`${flushType} is rendered`, () => {
          const component = render(<EuiButtonEmpty flush={flushType} />);

          expect(component).toMatchSnapshot();
        });
      });
    });

    describe('href', () => {
      it('secures the rel attribute when the target is _blank', () => {
        const component = render(<EuiButtonEmpty href="#" target="_blank" />);

        expect(component).toMatchSnapshot();
      });
    });

    describe('onClick', () => {
      it('supports onClick and href', () => {
        const handler = jest.fn();
        const component = mount(<EuiButtonEmpty href="#" onClick={handler} />);
        component.find('a').simulate('click');
        expect(handler.mock.calls.length).toEqual(1);
      });

      it('supports onClick as a button', () => {
        const handler = jest.fn();
        const component = mount(<EuiButtonEmpty onClick={handler} />);
        component.find('button').simulate('click');
        expect(handler.mock.calls.length).toEqual(1);
      });
    });

    test('contentProps is rendered', () => {
      const component = render(
        <EuiButtonEmpty contentProps={requiredProps}>Content</EuiButtonEmpty>
      );

      expect(component).toMatchSnapshot();
    });

    test('textProps is rendered', () => {
      const component = render(
        <EuiButtonEmpty textProps={requiredProps}>Content</EuiButtonEmpty>
      );

      expect(component).toMatchSnapshot();
    });
  });
});
