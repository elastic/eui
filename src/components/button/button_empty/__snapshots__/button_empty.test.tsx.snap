// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiButtonEmpty is rendered 1`] = `
<button
  aria-label="aria-label"
  class="euiButtonEmpty euiButtonEmpty--primary testClass1 testClass2"
  data-test-subj="test subject string"
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    >
      Content
    </span>
  </span>
</button>
`;

exports[`EuiButtonEmpty props color danger is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--danger"
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props color ghost is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--ghost"
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props color primary is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--primary"
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props color success is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--success"
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props color text is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--text"
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props color warning is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--warning"
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props contentProps is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--primary"
  type="button"
>
  <span
    aria-label="aria-label"
    class="euiButtonContent euiButtonEmpty__content testClass1 testClass2"
    data-test-subj="test subject string"
  >
    <span
      class="euiButtonEmpty__text"
    >
      Content
    </span>
  </span>
</button>
`;

exports[`EuiButtonEmpty props flush both is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--flushBoth"
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props flush left is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--flushLeft"
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props flush right is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--flushRight"
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props href secures the rel attribute when the target is _blank 1`] = `
<a
  class="euiButtonEmpty euiButtonEmpty--primary"
  href="#"
  rel="noopener noreferrer"
  target="_blank"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</a>
`;

exports[`EuiButtonEmpty props iconSide left is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--primary"
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonContent__icon"
      color="inherit"
      data-euiicon-type="user"
    />
    <span
      class="euiButtonEmpty__text"
    >
      Content
    </span>
  </span>
</button>
`;

exports[`EuiButtonEmpty props iconSide right is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--primary"
  type="button"
>
  <span
    class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
  >
    <span
      class="euiButtonContent__icon"
      color="inherit"
      data-euiicon-type="user"
    />
    <span
      class="euiButtonEmpty__text"
    >
      Content
    </span>
  </span>
</button>
`;

exports[`EuiButtonEmpty props iconType is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--primary"
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonContent__icon"
      color="inherit"
      data-euiicon-type="user"
    />
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props isDisabled is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty-isDisabled"
  disabled=""
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props isDisabled renders a button even when href is defined 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty-isDisabled"
  disabled=""
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props isDisabled renders if passed simply as disabled 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty-isDisabled"
  disabled=""
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props isLoading is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty-isDisabled"
  disabled=""
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiLoadingSpinner euiLoadingSpinner--medium euiButtonContent__spinner"
    />
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props isSelected is rendered as false 1`] = `
<button
  aria-pressed="false"
  class="euiButtonEmpty euiButtonEmpty--primary"
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props isSelected is rendered as true 1`] = `
<button
  aria-pressed="true"
  class="euiButtonEmpty euiButtonEmpty--primary"
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props size l is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--large"
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props size s is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--small"
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props size xs is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall"
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      class="euiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`EuiButtonEmpty props textProps is rendered 1`] = `
<button
  class="euiButtonEmpty euiButtonEmpty--primary"
  type="button"
>
  <span
    class="euiButtonContent euiButtonEmpty__content"
  >
    <span
      aria-label="aria-label"
      class="euiButtonEmpty__text testClass1 testClass2"
      data-test-subj="test subject string"
    >
      Content
    </span>
  </span>
</button>
`;
